### PR TITLE
Move ProtocolVersion to the Consensus class

### PR DESCRIPTION
**Pull request by dangershony**
The idea behind this PR is to make ProtocolVersion available per chain (right now versions of different chains are mixed together)
Also ProtocolVersion was defined in the Settings class which is incorrect as its a property of consensus the protocol.

### CODE REVIEW

File: src/External/NBitcoin/ConsensusFactory.cs [hunk 1]
```diff
@@ -69,40 +72,14 @@ protected bool IsTransaction<T>()
...
-        /// <summary>
-        /// A set of flags representing the capabilities of the protocol.
-        /// </summary>
-        /// <param name="protocolVersion">The version to build the flags from.</param>
-        /// <returns>The <see cref="ProtocolCapabilities"/>.</returns>
-        public virtual ProtocolCapabilities GetProtocolCapabilities(uint protocolVersion)
-        {
-            return new ProtocolCapabilities()
-            {
-                PeerTooOld = protocolVersion < 209U,
```
Comment (sondreb) [inline src/External/NBitcoin/ConsensusFactory.cs:91]: Have not reviewed everything yet, but is this "documentation" added elsewhere?

File: src/External/NBitcoin/TransactionBuilder.cs [hunk 2]
```diff
@@ -172,7 +170,7 @@ public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
             return result;
         }
 
-        #endregion
+        #endregion ICoinSelector Members
```
Comment (sondreb) [inline src/External/NBitcoin/TransactionBuilder.cs:173]: Regions are evil, always removed them.

File: src/Networks/Bitcoin/Blockcore.Networks.Bitcoin/BitcoinMain.cs [hunk 3]
```diff
@@ -66,8 +67,15 @@ public BitcoinMain()
                 [BitcoinBIP9Deployments.Segwit] = new BIP9DeploymentsParameters("Segwit", 1, 1479168000, 1510704000, BIP9DeploymentsParameters.DefaultMainnetThreshold)
             };
 
+            ConsensusProtocol consensusProtocol = new ConsensusProtocol()
+            {
+                ProtocolVersion = ProtocolVersion.WITNESS_VERSION,
+                MinProtocolVersion = ProtocolVersion.SENDHEADERS_VERSION,
```
Comment (sondreb) [inline src/Networks/Bitcoin/Blockcore.Networks.Bitcoin/BitcoinMain.cs:73]: Why is the MinProtocolVersion different than NodeSettings MinProtocolVersion in the Bitcoin CLI project?

File: src/Blockcore/P2P/Protocol/Message.cs [hunk 4]
```diff
@@ -95,14 +95,11 @@ public void ReadWrite(BitcoinStream stream)
             length = payloadBytes == null ? 0 : length;
             stream.ReadWrite(ref length);
 
-            if (stream.ProtocolVersion >= ProtocolVersion.MEMPOOL_GD_VERSION)
-            {
-                if (stream.Serializing)
-                    checksum = Hashes.Hash256(payloadBytes, 0, length).GetLow32();
+            if (stream.Serializing)
```
[the remark this answers is not in the document]
Comment (dangershony) [reply src/Blockcore/P2P/Protocol/Message.cs:98]: Agree this is old NBitcoin code but I will fix in this class.

File: src/External/NBitcoin/ConsensusFactory.cs (hunk 1 above)
Comment (dangershony) [reply src/External/NBitcoin/ConsensusFactory.cs:91]: ProtocolCapabilities is from NBitcoin and it was never used really so time to get rid of it

File: src/Networks/Bitcoin/Blockcore.Networks.Bitcoin/BitcoinMain.cs (hunk 3 above)
Comment (dangershony) [reply src/Networks/Bitcoin/Blockcore.Networks.Bitcoin/BitcoinMain.cs:73]: in reality `WITNESS_VERSION` and `SENDHEADERS_VERSION` are the same number (both at 70012) we should get rid of the wtiness one though (another PR)

**Pull request merged**